ivy/functional/backends/paddle/experimental/layers.py

```diff
@@ -56,7 +58,59 @@ def max_pool2d(
     ceil_mode: bool = False,
     out: Optional[paddle.Tensor] = None,
 ) -> paddle.Tensor:
-    raise IvyNotImplementedException()
+    dtype = x.dtype
+    x = x.astype("float32")
```

Comment:
this is not correct, you should only change the dtype if the current dtype was unsupported